UnleashClient/__init__.py

```diff
@@ -304,47 +292,58 @@ def initialize_client(self, fetch_toggles: bool = True) -> None:
                     )
 
                 if fetch_toggles:
-                    fetch_headers = {
+                    self.unleash_scheduler.start()
```

Comment:
I'm not sure this pattern of calling the scheduler start like this is going to work. That seems to raise an exception if it's already started. I see the check for the running state in the metrics but that looks like a hack to me and it's going to go wrong at some point. Can't we just call it once in this method like we did before?